api/v1/verticadb_types.go

```diff
@@ -840,6 +840,14 @@ type Proxy struct {
 	// use a secret path reference prefix, such as gsm://. Everything after the
 	// prefix is the name of the secret in the service you are storing.
 	TLSSecret string `json:"tlsSecret,omitempty"`
+}
+
+type ProxySubclusterConfig struct {
```

Comment:
Let's make `resources` a pointer. `resources` in the manifest even when you do not set it. Making a pointer will allow us to omit subclusters[].proxy from the manifest when proxy is disabled.

Reply:
Updated in https://github.com/vertica/vertica-kubernetes/pull/985/commits/201c6fff001e9faf91be18d9aa69184c3d8c7c76